scripts/apps/monitoring/directives/ItemActionsMenu.ts

```diff
@@ -59,43 +85,101 @@ export function ItemActionsMenu(superdesk, activityService, workflowService, arc
+                scope.menuGroups = [];
+
+                getActionsFromExtensions(item)
+                    .then((actionsFromExtensions) => {
+                        let intent = {action: 'list', type: getType(item)};
+                        let activitiesByGroupName: {[groupName: string]: Array<IActivity>} = {};
+
+                        // group activities by `activity.group`
+                        superdesk.findActivities(intent, item).forEach((activity: IActivity) => {
+                            if (workflowService.isActionAllowed(scope.item, activity.action)) {
+                                let group = activity.group ?? 'default';
+
+                                if (activitiesByGroupName[group] == null) {
+                                    activitiesByGroupName[group] = [];
+                                }
+
+                                activitiesByGroupName[group].push(activity);
+                            }
+                        });
+
+                        let menuGroups: Array<IAuthoringMenuGroup> = [];
+
+                        // take default menu groups, add activities and push to `menuGroups`
+                        AUTHORING_MENU_GROUPS.forEach((group) => {
+                            if (activitiesByGroupName[group._id]) {
+                                menuGroups.push({
+                                    _id: group._id,
+                                    label: group.label,
+                                    concate: group.concate,
+                                    actions: activitiesByGroupName[group._id]
+                                        .map((activity) => ({kind: 'activity-based', activity: activity})),
+                                });
+                            }
+                        });
+
+                        // go over `activitiesByGroupName` and add groups not present
+                        // in default groups (AUTHORING_MENU_GROUPS)
+                        Object.keys(activitiesByGroupName).forEach((groupName) => {
+                            var existingGroup = AUTHORING_MENU_GROUPS.find((g) => g._id === groupName);
+
+                            if (!existingGroup) {
+                                menuGroups.push({
+                                    _id: groupName,
+                                    label: groupName,
+                                    actions: activitiesByGroupName[groupName]
+                                        .map((activity) => ({kind: 'activity-based', activity: activity})),
+                                });
+                            }
+                        });
+
+                        // handle actions from extensions
+                        let extensionActionsByGroupName: {[groupName: string]: Array<IArticleAction>} = {};
+
+                        for (const action of actionsFromExtensions) {
+                            let name = action.groupId ?? 'default';
```

Comment:
just wondering why using `let` here and not `const`?

Reply:
I wasn't sure if `const` variables are block or function scoped and I knew that `let` is block scoped. ;D I've looked it up, `const` is block-scoped too, will update it.